File: php/005/main.php
<?php
/*
 * ProjectEuler 005
 * PHP / main.php
 * copyright (c) 2015 Susisu
 *
 * PHP 5.4.30
 * `php main.php`
 */

function main() {
    $prod = 1;
    for ($n = 1; $n <= 20; $n++) {
        $prod *= $n / gcd($prod, $n);
    }
    echo "$prod\n";
}

function gcd($a, $b) {
    if ($a < $b) {
        return gcd($b, $a);
    }
    else if($b === 0) {
        return $a;
    }
    else {
        return gcd($b, $a % $b);
    }
}

main();

File: python/005/main.py
# ProjectEuler 005
# Python / main.py
# copyright (c) 2015 Susisu
#
# Python 3.4.2
# `python3 main.py`

from fractions import gcd

def main():
    p = 1
    for i in range(1, 21):
        p *= i // gcd(p, i)
    print(p)

main()
